dor/providers/package_generator.py

```diff
@@ -52,6 +53,39 @@ def serialize_provenance(data: Any) -> str:
     return xmldata
 
 
+class FileSetsPendingError(Exception):
+
+    def __init__(self, message: str):
+        super().__init__(message)
+
+        self.message = message
+
+
+class FileSetsPending:
+    def __init__(self, file_provider: FileProvider, root_path: Path):
+        self.file_provider = file_provider
+        self.root_path = root_path
+
+    def pending(self, file_set_id: str) -> bool:
+        return (self.root_path / file_set_id).exists()
+
+    def push(self, file_set_id: str, destination_path: Path) -> None:
+        if not self.pending(file_set_id):
+            raise FileSetsPendingError(
+                f"Expected file set \"{file_set_id}\" to be pending"
+            )
+
+        file_set_directories = [
+            entry.name for entry in sorted((self.root_path / file_set_id).iterdir(), key=os.path.getmtime, reverse=True)
+            if entry.is_dir()
+        ]
+
+        self.file_provider.clone_directory_structure(
+            self.root_path / file_set_id / file_set_directories[0],
+            destination_path / file_set_id
+        )
```

Comment:
Don't we also need to look in the `build` directory? I think fileset directories that can be copied should be stored at `{fileset id} / {ordinal number} / build / {fileset id}`